vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt

```diff
@@ -71,6 +76,17 @@ class VectorSettingsPreferencesFragment @Inject constructor(
             }
         }
 
+        findPreference<VectorSwitchPreference>(LightweightSettingsStorage.MATRIX_SDK_SETTINGS_PRESENCE_OFFLINE_MODE_ENABLED)!!.let { pref ->
+            pref.isChecked = lightweightSettingsStorage.getPresenceOfflineModeEnabled()
+            pref.setOnPreferenceChangeListener { _, newValue ->
+                val presenceOfflineModeEnabled = newValue as? Boolean ?: false
+                lifecycleScope.launch {
```

Comment:
I wonder if there is a better place to start this but right now we don't have a viewModel or interactor on Preferences